src/dhtrunner.cpp

```diff
@@ -531,17 +534,35 @@ DhtRunner::startNetwork(const SockAddr sin4, const SockAddr sin6)
 
     s4 = -1;
     s6 = -1;
+    DhtException inetBindException, inet6BindException;
 
     bound4 = {};
-    if (sin4)
-        s4 = bindSocket(sin4, bound4);
+    if (sin4) {
+        try {
+            s4 = bindSocket(sin4, bound4);
+        } catch (const DhtException& e) {
+           inetBindException = e;
```

Comment:
just print the socket-specific error there and throw a generic "can't bind socket" exception later ?